test/functional/wallet_avoidreuse.py

```diff
@@ -0,0 +1,217 @@
+    }
+    supports_reused = True
+    for utxo in node.listunspent(minconf=0):
+        r["total"]["count"] += 1
+        r["total"]["sum"] += utxo["amount"]
+        if supports_reused and "reused" in utxo:
+            if utxo["reused"]:
+                r["reused"]["count"] += 1
+                r["reused"]["sum"] += utxo["amount"]
+        else:
+            supports_reused = False
+    r["reused"]["supported"] = supports_reused
+    return r
+
+def assert_unspent(node, total_count=None, total_sum=None, reused_supported=None, reused_count=None, reused_sum=None):
+    '''Make assertions about a node's unspent output statistics'''
+    stats = count_unspent(node)
+    if total_count is not None:
+        assert_equal(stats["total"]["count"], total_count)
+    if total_sum is not None:
+        assert_approx(stats["total"]["sum"], total_sum, 0.001)
+    if reused_supported is not None:
+        assert_equal(stats["reused"]["supported"], reused_supported)
+    if reused_count is not None:
+        assert_equal(stats["reused"]["count"], reused_count)
+    if reused_sum is not None:
+        assert_approx(stats["reused"]["sum"], reused_sum, 0.001)
+
+class AvoidReuseTest(BitcoinTestFramework):
+
+    def set_test_params(self):
+        self.setup_clean_chain = False
+        self.num_nodes = 2
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def run_test(self):
+        '''Set up initial chain and run tests defined below'''
+
+        self.test_persistence()
+        self.test_immutable()
+
+        self.nodes[0].generate(110)
+        self.sync_all()
+        reset_balance(self.nodes[1], self.nodes[0].getnewaddress())
+        self.test_fund_send_fund_senddirty()
+        reset_balance(self.nodes[1], self.nodes[0].getnewaddress())
+        self.test_fund_send_fund_send()
+
+    def test_persistence(self):
+        '''Test that wallet files persist the avoid_reuse flag.'''
+        # Configure node 1 to use avoid_reuse
+        self.nodes[1].setwalletflag('avoid_reuse')
+
+        # Flags should be node1.avoid_reuse=false, node2.avoid_reuse=true
+        assert_equal(self.nodes[0].getwalletinfo()["avoid_reuse"], False)
+        assert_equal(self.nodes[1].getwalletinfo()["avoid_reuse"], True)
+
+        # Stop and restart node 1
+        self.stop_node(1)
+        self.start_node(1)
+        connect_nodes_bi(self.nodes, 0, 1)
+
+        # Flags should still be node1.avoid_reuse=false, node2.avoid_reuse=true
+        assert_equal(self.nodes[0].getwalletinfo()["avoid_reuse"], False)
+        assert_equal(self.nodes[1].getwalletinfo()["avoid_reuse"], True)
+
+        # Attempting to set flag to its current state should throw
+        assert_raises_rpc_error(-8, "Wallet flag is already set to false", self.nodes[0].setwalletflag, 'avoid_reuse', False)
+        assert_raises_rpc_error(-8, "Wallet flag is already set to true", self.nodes[1].setwalletflag, 'avoid_reuse', True)
+
+    def test_immutable(self):
```

Comment:
In addition to the above TODO, I also suggest this be moved to a different file later (its independent of the avoid-reuse tests)

Reply:
Makes sense to me.